File: Python/Reverse Words in a String.py
class Solution:
    # @param s, a string
    # @return a string
    def reverseWords(self, s):
        stack = []
        word = []
        lastisblank = True
        for c in s:
            if c != " ":
                word.append(c)
                lastisblank = False
            elif lastisblank and c == " ":
                pass
            else:
                stack.append(word)
                word = []
                lastisblank = True
        if word:
            stack.append(word)
            
        reverseword = []
        for w in stack[::-1]:
            reverseword.append("".join(w))
            
        ret =  " ".join(reverseword)       
        return ret

File: Java/src/Reverse Words in a String.java
public class Solution {
    public String reverseWords(String s) {
        String[] parts = s.trim().split("\\s+");
        String ret = "";
        if (parts.length > 0) {
            for (int i = parts.length - 1; i > 0; i--) {
                ret += parts[i] + " ";
            }
            ret += parts[0];
        }
        return ret;
        
    }
}